### ISSUE TITLE
USWDS-Site - Feature: Update guidance for table component caption

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds-site#3100]
### Is your feature request related to a problem? Please describe.

Related to [#6253](https://github.com/uswds/uswds/issues/6253) and [#489](https://github.com/uswds/uswds-team/issues/489)

We should update the guidance on the [table page](https://designsystem.digital.gov/components/table/) to specify the importance of the `caption` element and its relationship to a table. It serves as the title of the table and we should encourage people to use the element in almost all cases.

Current guidance on our [table page](https://designsystem.digital.gov/components/table/):

Add title and attribution in a caption. When adding a title, attribution, or a last-updated date to a table, include it in the <caption> tag inside of the <table> element.

### Describe the solution you'd like

Draft proposed change:

Use a  `caption`  to describe the purpose of the table. When adding a title, attribution, or a last-updated date to a table, include it in the <caption> tag inside of the `table` element. Keep the caption wording short.

Ensure sighted users are able to identify the caption as the title of the table. Make the caption style larger than the body style of the table text. Do not match an existing heading level or other style (such as an H2 or H3). Caption styles should be noticeable yet distinct.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate feature requests.

**Comment by dct2023** [comment]
@amycole501 to share the draft version to have a full scope of this request.

cc: @mejiaj @amyleadem
